Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -1572,21 +1600,56 @@ else if (equipSlotArr.length>1)
 			boolean swapSlot = CommonChecks.checkCharacterIsPlayer(character) && 
 					("Shield".equals(equipment.getProperty("itemType")) || "Weapon".equals(equipment.getProperty("itemType")));
 			
-			for(int i = equipSlotArr.length-1; i>=0; i--)
-			{
+			//this loop iterates backwards through equipSlotArr to find the first entry that is valid. If there are none, toss an error.
+			for(int i = equipSlotArr.length-1; i>=0; i--) {	
+				
 				String slotCheck = equipSlotArr[i].trim();
 				
-				if(CommonChecks.checkIsValidEquipSlot(slotCheck)==false) 
+				//we split the destination slot at " and ". if there are more than 1 results from that, it's a complex item.
+				String[]complexSlot = slotCheck.split(" and ");
+				
+				if(complexSlot.length > 1) {
+					boolean isValid = true;
+					
+					//this loop iterates through complexSlot and sees if all those slots are open. If a single one isn't, we break out.
+					for(int k=0; k!=complexSlot.length; k++) {
+						complexSlot[i] = complexSlot[i].trim(); //clean up the string, just in case.
+						
+						//if the equip slot doesn't exist, we break.
+						if(CommonChecks.checkIsValidEquipSlot(complexSlot[i]) == false) {
+							isValid = false;
+							break;
+						}
+						
+						//if its 2hands, fix it.
+						if(complexSlot[i].equals("2Hands")) {
+							complexSlot[i] = "LeftHand and RightHand";
+						}
+
+						
+						//if there is something equipped in one of the slots, we break.
+						if(character.getProperty("equipment" + complexSlot[i]) != null){
+							isValid = false;
+							break;
+						}
```

Comment:
In this scenario, `2Hands` slot will _never_ work in a complex slot, since you'll be checking for `equipmentLeftHand and RightHand` property on character entity.